src/Storages/MergeTree/MergeFromLogEntryTask.cpp

```diff
@@ -357,6 +357,13 @@ bool MergeFromLogEntryTask::finalize(ReplicatedMergeMutateTaskBase::PartLogWrite
 
             ProfileEvents::increment(ProfileEvents::DataAfterMergeDiffersFromReplica);
 
+            Strings files_with_size;
```

Comment:
I wonder how useful this is in reality. For example, if you get an exception like https://github.com/ClickHouse/ClickHouse/issues/34595 (issue where this message appeared), how useful is to know the list of local files?

Would it be possible to provide an actual / proper comparison against the ZK part so you know if it was a compression change, different file content (and which one), etc?

Reply:
> I wonder how useful this is in reality. For example, if you get an exception like https://github.com/ClickHouse/ClickHouse/issues/34595 (issue where this message appeared), how useful is to know the list of local files?

Unlikely it will be very useful for users, but if I will see `cmrk3` or `serializations.json` I would definitely understand one possible root case.

But experienced users can also guess I would say (at least he can go to a replica and look at the file, and/or google this).

What do you think?

> Would it be possible to provide an actual / proper comparison against the ZK part so you know if it was a compression change, different file content (and which one), etc?

Hm, I don't see how it is possible, since in case of compact header there is only hashes.

Reply:
Yep, you are right.

Maybe we can extend system tables (system.parts_columns? system.parts? system.part_files?) so it's possible to get that same information from the new error message in the other server, compare and verify